PopPUNK/network.py

```diff
@@ -1837,3 +1888,68 @@ def sparse_mat_to_network(sparse_mat, rlist, use_gpu = False):
                                                summarise=False)
 
     return G
+
+
+def prune_graph(prefix, reflist, passed, output_db_name, threads, use_gpu):
+    """Keep only the specified sequences in a graph
+
+    Args:
+       prefix (str)
+           Name of directory containing network
+       reflist (list)
+           Ordered list of sequences of database
```

Comment:
I don't like that this has to be ordered, or I don't understand it. Why is this needed? Seems easy to get wrong

Reply:
Needs to be ordered because graph nodes are just integers. I did manually check this was working on a toy example as I was also nervous. This is only called within `remove_qc_fail`, which uses the ordered `names` for the distance matrix pruning, so hopefully limited scope for error beyond our current implementation.

Reply:
Can we at least add a check that it is ordered in the function?